tests/pytests/unit/utils/test_cloud.py

```diff
@@ -0,0 +1,230 @@
+"""
+    :codeauthor: Pedro Algarvio (pedro@algarvio.me)
+    tests.unit.utils.cloud_test
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Test the salt-cloud utilities module.
+
+"""
+
+
+import os
+import shutil
+import tempfile
+
+import pytest
+import salt.utils.cloud as cloud
+from tests.support.mock import MagicMock, patch
+from tests.support.runtests import RUNTIME_VARS
+from tests.support.unit import SkipTest
+
+
+@pytest.fixture()
+def configure_loader_modules():
+    return {
+        cloud: {"__opts__": {"sock_dir": "master"}},
+    }
+
+
+@pytest.fixture()
+def create_class():
+    old_cwd = os.getcwd()
+    gpg_keydir = os.path.join(RUNTIME_VARS.TMP, "gpg-keydir")
+    try:
+        # The keyring library uses `getcwd()`, let's make sure we in a good directory
+        # before importing keyring
+        if not os.path.isdir(gpg_keydir):
+            os.makedirs(gpg_keydir)
+        os.chdir(gpg_keydir)
+
+        # Late import because of the above reason
+        import keyring
+        import keyring.backend
+
+        class CustomKeyring(keyring.backend.KeyringBackend):
+            """
+            A test keyring which always outputs same password
+            """
+
+            def __init__(self):
+                self.__storage = {}
+
+            def supported(self):
+                return 0
+
+            def set_password(
+                self, servicename, username, password
+            ):  # pylint: disable=arguments-differ
+                self.__storage.setdefault(servicename, {}).update({username: password})
+                return 0
+
+            def get_password(
+                self, servicename, username
+            ):  # pylint: disable=arguments-differ
+                return self.__storage.setdefault(servicename, {}).get(username, None)
+
+            def delete_password(
+                self, servicename, username
+            ):  # pylint: disable=arguments-differ
+                self.__storage.setdefault(servicename, {}).pop(username, None)
+                return 0
+
+        # set the keyring for keyring lib
+        keyring.set_keyring(CustomKeyring())
+        yield
+    except ImportError:
+        raise SkipTest('The "keyring" python module is not installed')
+    finally:
+        os.chdir(old_cwd)
+        if os.path.exists(gpg_keydir):
+            shutil.rmtree(gpg_keydir)
+        del gpg_keydir
```

Comment:
```suggestion
@pytest.fixture
def create_class(tmp_path):
    old_cwd = os.getcwd()
    gpg_keydir = tmp_path / "gpg-keydir"
    gpg_keydir.mkdir()
    try:
        # The keyring library uses `getcwd()`, let's make sure we in a good directory
        # before importing keyring
        os.chdir(str(gpg_keydir))

        # Late import because of the above reason
        import keyring
        import keyring.backend

        class CustomKeyring(keyring.backend.KeyringBackend):
            """
            A test keyring which always outputs same password
            """

            def __init__(self):
                self.__storage = {}

            def supported(self):
                return 0

            def set_password(
                self, servicename, username, password
            ):  # pylint: disable=arguments-differ
                self.__storage.setdefault(servicename, {}).update({username: password})
                return 0

            def get_password(
                self, servicename, username
            ):  # pylint: disable=arguments-differ
                return self.__storage.setdefault(servicename, {}).get(username)

            def delete_password(
                self, servicename, username
            ):  # pylint: disable=arguments-differ
                self.__storage.setdefault(servicename, {}).pop(username, None)
                return 0

        # set the keyring for keyring lib
        keyring.set_keyring(CustomKeyring())
        yield
    except ImportError:
        pytest.skip('The "keyring" python module is not installed')
    finally:
        os.chdir(old_cwd)
```